src/catnip/runtime/memory/manager.rs

```diff
@@ -198,10 +196,11 @@ impl MemoryManager {
         } else {
             // Release heap-managed buffer.
             let sgaseg: dmtr_sgaseg_t = sga.sga_segs[0];
-            let (ptr, len): (*mut c_void, usize) = (sgaseg.sgaseg_buf, sgaseg.sgaseg_len as usize);
-            let allocation: Box<[u8]> =
-                unsafe { Box::from_raw(slice::from_raw_parts_mut(ptr as *mut _, len)) };
-            drop(allocation);
+            let (data_ptr, length): (*mut u8, usize) =
+                (sgaseg.sgaseg_buf as *mut u8, sgaseg.sgaseg_len as usize);
+
+            // Convert back raw slice to a heap buffer and drop allocation.
```

Comment:
This technically isn't a Rust "slice", but rather just a pointer length pair, correct?

Reply:
This is technically a "raw part", since it was produced by a previous call to `slice::from_raw_parts()`. Please be clear on the action item on this.

Reply:
I was referring to the use of the term "slice" in the comment.  This function has no idea where the pointer length pair came from, just that it was in the scatter-gather array that it's freeing.  There's an assumption here that it came from the same memory allocator that the DataBuffer type uses, as it converts it back into one just to free it.  I was going to comment on that elsewhere.

But for now, I'd just get rid of the term "slice" in the comment.

Reply:
Sure. It is fixed.